rust/worker/src/blockstore/arrow/blockfile.rs

```diff
@@ -142,13 +183,19 @@ impl ArrowBlockfileWriter {
                 let block = self.block_manager.get(&target_block_id).await.unwrap();
                 let new_delta = self.block_manager.fork::<K, V>(&block.id);
                 let new_id = new_delta.id;
-                self.sparse_index.replace_block(
-                    target_block_id,
-                    new_delta.id,
-                    new_delta
-                        .get_min_key()
-                        .expect("Block should never be empty when forked"),
-                );
+                // Blocks can be empty.
+                if new_delta.len() == 0 {
+                    self.sparse_index
+                        .replace_lone_block(target_block_id, new_delta.id);
+                } else {
+                    self.sparse_index.replace_block(
+                        target_block_id,
+                        new_delta.id,
+                        new_delta
+                            .get_min_key()
```

Comment:
we can remove get_min_key() I think. And we can change replace block to reuse the min key - then we can remove replace_lone_block